client/src/index.jsx

```diff
@@ -60,17 +60,17 @@ class App extends React.Component {
     this.cache[cacheName].set(productId, data);
   }
 
-  updateReviews(reviews) {
+  updateReviews(reviewsData) {
     this.setState({
-      selectedProductReviews: reviews,
+      selectedProductReviews: { reviewsData: reviewsData, reviews: reviewsData.results }
```

Comment:
This will force re-rendering unless you write custom `shouldComponentUpdate` methods to compare objects by value instead of by reference.

Reply:
I've noticed the re-renders too and i'm checking my files just in case. 

If we are storing ratings and reviews in cache for the entire app, I don't think there's a need for this state. Would you prefer passing down cache? Or setting this state to reference the cache, and pass the state down instead?